pygmt/src/triangulate.py

```diff
@@ -0,0 +1,381 @@
+        registration=None,
+        skiprows=None,
+        wrap=None,
+        **kwargs,
+    ):
+        """
+        Delaunay triangle based gridding of Cartesian data.
+
+        Reads in x,y[,z] data and performs Delaunay triangulation, i.e., it
+        finds how the points should be connected to give the most equilateral
+        triangulation possible. If a map projection (give ``region`` and
+        ``projection``) is chosen then it is applied before the triangulation
+        is calculated. By setting ``outgrid`` and ``spacing``, a grid will be
+        calculated based on the surface defined by the planar triangles. The
+        actual algorithm used in the triangulations is either that of Watson
+        [1982] or Shewchuk [1996] [Default is Shewchuk if installed; type
+        ``gmt get GMT_TRIANGULATE`` on the command line to see which method is
+        selected). This choice is made during the GMT installation.
+        Furthermore, if the Shewchuk algorithm is installed then you can also
+        perform the calculation of Voronoi polygons and optionally grid your
+        data via the natural nearest neighbor algorithm. **Note**: For
+        geographic data with global or very large extent you should consider
+        :gmt-docs:`sphtriangulate <sphtriangulate.html>` instead since
+        ``triangulate`` is a Cartesian or small-geographic area operator and is
+        unaware of periodic or polar boundary conditions.
+
+        Must provide either ``data`` or ``x``, ``y``, and ``z``.
+
+        Must provide ``region`` and ``spacing``.
+
+        Full option list at :gmt-docs:`triangulate.html`
+
+        Parameters
+        ----------
+        x/y/z : np.ndarray
+            Arrays of x and y coordinates and values z of the data points.
+        data : str or {table-like}
+            Pass in (x, y, z) or (longitude, latitude, elevation) values by
+            providing a file name to an ASCII data table, a 2D
+            {table-classes}.
+        {J}
+        {R}
+        {I}
+        outgrid : str or None
+            The name of the output netCDF file with extension .nc to store the
+            grid in. The interpolation is performed in the original
+            coordinates, so if your triangles are close to the poles you are
+            better off projecting all data to a local coordinate system before
+            using ``triangulate`` (this is true of all gridding routines) or
+            instead select :gmt-docs:`sphtriangulate <sphtriangulate.html>`.
+        {V}
+        {b}
+        {d}
+        {e}
+        {f}
+        {h}
+        {i}
+        {r}
+        {s}
+        {w}
+
+        Returns
+        -------
+        ret: xarray.DataArray or None
+            Return type depends on whether the ``outgrid`` parameter is set:
+
+            - xarray.DataArray if ``outgrid`` is None (default)
+            - None if ``outgrid`` is a str (grid output is stored in
+              ``outgrid``)
+        """
```

Comment:
```suggestion

        Notes
        -----
        For geographic data with global or very large extent you should
        consider :gmt-docs:`sphtriangulate <sphtriangulate.html>` instead since
        ``triangulate`` is a Cartesian or small-geographic area operator and is
        unaware of periodic or polar boundary conditions.
        """
```